### ISSUE TITLE
[bug] GNN training accepts validation data without evaluating it

### ISSUE THREAD

**Issue by ThunderKhan** [KanishJebaMathewM/Truxify#15902]
**Describe the bug**

`RouteOptimizer.train()` accepts a `val_data` argument, but the current training implementation does not use it during the training process.

The method computes training loss from `train_data` and returns the final training loss, while the supplied validation dataset is never evaluated. This makes the validation parameter effectively non-functional and prevents callers from detecting overfitting or monitoring generalization during training.

The current flow is effectively:

```python
RouteOptimizer.train(train_data, val_data, epochs)
        ↓
iterate over train_data
        ↓
calculate training loss
        ↓
return avg_loss
```

The `val_data` argument is accepted but does not contribute to any validation metric.

**To Reproduce**

1. Create a valid `RouteOptimizer`.
2. Prepare a non-empty training dataset.
3. Prepare a separate validation dataset.
4. Call:

```python
optimizer.train(
    train_data=train_data,
    val_data=val_data,
    epochs=2,
)
```

5. Observe that training completes using `train_data`.
6. Observe that no validation loss or validation metric is computed from `val_data`.
7. Compare the returned result and optimizer state before and after supplying validation data.

The behavior is the same whether meaningful validation data is supplied or the validation dataset is omitted, apart from the fact that `val_data` is accepted by the method signature.

**Expected behavior**

When validation data is supplied, `RouteOptimizer.train()` should evaluate the model against that dataset after each training epoch or at another clearly defined validation interval.

Validation should:

* run without gradient updates;
* use evaluation mode so training-only behavior such as dropout does not affect the validation calculation;
* compute and retain a validation loss or equivalent metric;
* avoid modifying model weights during validation;
* remain optional when `val_data` is not supplied.

Training behavior for `train_data` should remain unchanged.

The implementation should also make the distinction between training and validation metrics explicit rather than silently accepting a validation dataset that is never consumed.

**Screenshots**

Not applicable. This is a backend ML training issue.

**Environment (please complete the following information):**

* OS: Windows
* App Version: Current development version
* Browser: N/A
* Python: 3.12
* Component: `backend/ml/gnn`
* Affected method: `RouteOptimizer.train()`
* Test framework: `pytest`
* Framework: PyTorch / PyTorch Geometric

**Additional context**

A `val_data` parameter exists in the training API, which indicates that validation is intended to be part of the training workflow. However, ignoring the supplied validation set makes the training interface misleading and removes an important mechanism for monitoring model generalization.

The fix should remain narrowly scoped to validation evaluation. It should not introduce a new training algorithm or alter the existing optimizer, loss function, or model architecture.

Regression coverage should verify:

* validation data is evaluated when supplied;
* a validation metric is recorded for each training epoch;
* validation runs under `torch.no_grad()`;
* validation does not update model parameters;
* training still works when no validation data is supplied;
* existing GNN training and routing tests continue to pass.

**Comment by ThunderKhan** [comment]
I’m claiming this issue and will be implementing the fix on my branch. I’ll open a PR with the validation evaluation changes and regression coverage in a few minutes.
